index.bs

```diff
@@ -1658,6 +1677,37 @@ and a [=response=] |response|:
 1. Run [=process an attribution eligible response=] with |contextOrigin|,
     |eligibility|, and |response|.
 
+To <dfn>get the registration platform</dfn> given |webHeader|,
+|osRegistrations|, and |preferredPlatform|:
+
+1. If |webHeader| and |osRegistrations| are both null, return null.
+1. If |preferredPlatform| is null:
+    1. If |webHeader| and |osRegistrations| are both not null, return null.
+    1. If |webHeader| is not null and the user agent supports web registrations,
+        return "<code>[=registrar/web=]</code>".
+    1. if |osRegistrations| is not null and the user agent supports OS registrations,
+        return "<code>[=registrar/os=]</code>".
+    1. Return null.
+1. If |preferredPlatform| is:
+    <dl class="switch">
+      : "<code>[=registrar/web=]</code>"
+      ::
+         1. If |webHeader| is null, return null.
+         1. If the user agent supports web registrations, return "<code>[=registrar/web=]</code>".
+         1. If |osRegistrations| is not null and the user agent supports OS registrations,
+             return "<code>[=registrar/os=]</code>".
+         1. Return null.
+
+      : "<code>[=registrar/os=]</code>"
+      ::
+         1. If |osRegistrations| is null, return null.
+         1. If the user agent supports OS registrations, return "<code>[=registrar/os=]</code>".
+         1. If |webHeader| is not null and the user agent supports web registrations,
+             return "<code>[=registrar/web=]</code>".
+         1. Return null.
+
+    </dl>
+
 To <dfn export>process an attribution eligible response</dfn> given a [=suitable origin=]
```

Comment:
I think this algorithm has gotten a bit hard to follow given its length. Can we split out some of the subalgorithms?